pkg/utils/addons/addons.go

```diff
@@ -121,26 +142,65 @@ func CreatePipelines(template mf.Manifest, client client.Client) (mf.Manifest, e
 			pipelineParams = append(pipelineParams.([]interface{}), map[string]interface{}{"name": "MINOR_VERSION", "type": "string"})
 		}
 		if err := unstructured.SetNestedField(newTempRes.Object, pipelineParams, "spec", "params"); err != nil {
-			return mf.Manifest{}, err
+			return nil, err
 		}
 
 		if err := unstructured.SetNestedField(taskBuild, taskParams, "params"); err != nil {
-			return mf.Manifest{}, nil
+			return nil, nil
 		}
 
 		//adding the deploy task
 		for _, tg := range taskGenerators {
 			p, err := tg.generate(newTempRes)
```

Comment:
suggestion: if we consider all kinds of modifcations to pipelines as a modifier (`type pipelineModifier func(map[string]interface{}) map[string]interface{}`), then we could use a second loop to take care of various modifications here.